src/js/contexts/MessageContext/MessageContext.js

```diff
@@ -32,11 +32,26 @@ export const format = (options, messages) => {
       messageObj = messageObj[idPart];
     }
   });
+
   const message =
     (options.messages ? options.messages[baseId] : undefined) ||
     messageObj ||
     options.defaultMessage;
-  return message;
+
+  const values = options.values ? options.values : undefined;
+
+  let newMessage;
+  if (values) {
+    newMessage = message.split(' ');
```

Comment:
your right bad assumption! l'll check this out

Reply:
Hey @MikeKingdom , could you provide an example of this as well so I can get a better view of what we might not be accounting for?

Reply:
Surely. Ideographic languages like Japanese, Chinese, Korean, etc, often don't use spaces to separate words or phrases. Here's what the "Moved to {date}" string would be translated to in Japanese:
`"{date}に移動しました"`
There are no spaces there not to mention the date appears first (hence the need for substitution rather than concatenation in the code).

Reply:
And one thought that just occurred to me is it might be safest to not replace tokens we think we find but which don't have a corresponding thing in `values`. My example code would just replace them with an empty string.

Reply:
So if we don't have a corresponding `values`  we should not replace them with an empty string ?

Reply:
That would be my suggestion I think simply so that if our way of processing substitutions tries to replace something the developer didn't intend as a parameter it would just leave it there. That kind of thing seems more likely with backend error content and such that we wouldn't control (although probably not the case with Grommet itself. However. this also helps people notice if they have a message where they missed providing a value.

Reply:
BTW. I would still recommend something more like my suggested code above for parsing the substitutions.

Reply:
got it thanks for clarifying!/l